simulations/NavierStokes_2D/scripts/train_posteriori.jl

```diff
@@ -13,15 +13,41 @@ rng = Random.Xoshiro(123)
 ig = 1 # index of the LES grid to use.
 include("preprocess_posteriori.jl")
 
-# * Creation of the model: NN closure
-closure, θ, st = cnn(;
-    setup = setups[ig],
+using CoupledNODE: AttentionLayer
+using ComponentArrays: ComponentArray
+using Lux: Lux
+u = io_post[ig].u[:, :, :, 1, 1:50]
+T = setups[1].T
+d = D = setups[1].grid.dimension()
+N = size(u, 1)
+emb_size = 8
+patch_size = 3
+n_heads = 2
+
+# * Define the CNN layers
+# since I will use them after the attention (that gets concatenated with the input), I have to start from 2*D channels
+CnnLayers, _, _ = cnn(;
+    T = T,
+    D = D,
+    data_ch = 2 * D,
     radii = [3, 3],
     channels = [2, 2],
     activations = [tanh, identity],
     use_bias = [false, false],
     rng
 )
+layers = (
+    Lux.SkipConnection(AttentionLayer(N, d, emb_size, patch_size, n_heads; T = T),
+        (x, y) -> cat(x, y; dims = 3); name = "Attention"),
+    CnnLayers
+)
+closure = Lux.Chain(layers...)
+θ, st = Lux.setup(rng, closure)
+using ComponentArrays: ComponentArray
+θ = ComponentArray(θ)
```

Comment:
Why do we need this additional block of code?
it seems to me that we should bstract this in the creation of `cnn` or propose another model `cnn_with_attention`. In any case, the same code for generating the model has to be run in the `evaluate_posteriori.jl` so that when loading the model we can apply the weights, avoiding incompatibilities.